device-registry/src/main/scala/org/genivi/sota/device_registry/db/DeviceRepository.scala

```diff
@@ -97,13 +98,17 @@ object DeviceRepository {
   }
 
   def updateLastSeen(uuid: Uuid)
```

Comment:
IMO this should be written as:

```
  def updateLastSeen(uuid: Uuid)
                    (implicit ec: ExecutionContext): DBIO[Unit] = {

    val now = Instant.now()

    val dbIO = for {
      _ <- devices.filter(_.uuid == uuid).filter(_.activatedAt.isEmpty).map(_.activatedAt).update(Some(now))
      _ <- devices.filter(_.uuid == uuid).map(_.lastSeen).update(Some(now))
    } yield ()

    dbIO.transactionally
  }
```
Also, it should never return a `DeviceSeen`. Device repository is a database encapsulation object, as the  name suggests. DeviceSeen is a BusMessage, a database should not know anything about bus.